warehouse/slave.go

```diff
@@ -35,140 +41,162 @@ const (
 	statsWorkerClaimProcessingTime      = "worker_claim_processing_time"
 	statsWorkerClaimProcessingFailed    = "worker_claim_processing_failed"
 	statsWorkerClaimProcessingSucceeded = "worker_claim_processing_succeeded"
-	tagWorkerid                         = "workerId"
-)
 
-const (
+	tagWorkerid = "workerId"
+
 	workerProcessingDownloadStagingFileFailed = "worker_processing_download_staging_file_failed"
 )
 
-// JobRun Temporary store for processing staging file to load file
-type JobRun struct {
+type uploadProcessingResult struct {
+	result uploadResult
+	err    error
+}
+
+type uploadResult struct {
+	TableName             string
+	Location              string
+	TotalRows             int
+	ContentLength         int64
+	StagingFileID         int64
+	DestinationRevisionID string
+	UseRudderStorage      bool
+}
+
+type asyncJobRunResult struct {
+	Result bool
+	Id     string
```

Comment:
[minor]
```suggestion
	ID     string
```
We can use JSON tag if we marshal this